power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -12,6 +12,8 @@
 #include "dataset_fwd.hpp"
 #include "meta_data.hpp"
 
+#include <boost/range.hpp>
```

Comment:
do you still use this header? if so, you can keep it

Reply:
I need it for `boost::make_iterator_range`. However, it may be removed after the possible bug you found. I'll update this here.

Reply:
The bug you have encountered does not affect the use of `boost::make_iterator_range`. So `<boost/range.hpp>` will stay.